src/core/router.js

```diff
@@ -231,9 +233,16 @@ var router = {
                         window.require(
                             dependencies,
                             function() {
+                                if(_.has(self.activeController, 'destroy')){
+                                    self.activeController.destroy();
+                                }
+
+                                self.trigger('destroy');
+
```

Comment:
The `destroy` event should only be triggered when a controller is active and is actually destroyed.
And the variable must be voided as it may not be changed if no controller matches.

```suggestion
                                if (activeController) {
                                    if (_.has(self.activeController, 'destroy')) {
                                        activeController.destroy();
                                    }
                                    activeController = null;
                                    self.trigger('destroy');
                                }
```

Reply:
Refactored